fasthttputil/pipeconns.go

```diff
@@ -158,6 +160,8 @@ func (c *pipeConn) readNextByteBuffer(mayBlock bool) error {
 		if !mayBlock {
 			return errWouldBlock
 		}
+		c.readDeadlineChLock.Lock()
+		defer c.readDeadlineChLock.Unlock()
```

Comment:
I'm just wondering if this can cause a locking issue with the channel reads below taking to long. Wouldn't it be better to only do `readDeadlineCh := c.readDeadlineCh` within a mutex and read from that below?

Reply:
@erikdubbelboer That is absolutely right. I have made the fix and added a new commit. Thank you.